app/controllers/two_factor_authentication/recovery_code_verification_controller.rb

```diff
@@ -10,11 +10,25 @@ def create
 
       analytics.track_event(Analytics::MULTI_FACTOR_AUTH, result.merge(method: 'recovery code'))
 
+      handle_result(result)
+    end
+
+    private
+
+    def handle_result(result)
       if result[:success]
+        re_encrypt_profile_recovery_pii if current_user.password_reset_profile.present?
         handle_valid_otp
       else
         handle_invalid_otp(type: 'recovery_code')
       end
     end
+
+    def re_encrypt_profile_recovery_pii
+      profile = current_user.password_reset_profile
```

Comment:
perhaps we can have `profile` and `pii` methods separately (lower in this file)? functionally will be the same, but I think it would improve readability of this method.